src/modules/consumer.js

```diff
@@ -155,6 +122,84 @@ class Consumer {
         }
       });
   }
+
+  _consumeQueue(channel, queue, callback) {
+    return channel
+      .consume(
+        queue,
+        (msg) => {
+          if (!msg) {
+            // When forcefully cancelled by rabbitmq, consumer would receive a null message.
+            // https://amqp-node.github.io/amqplib/channel_api.html#channel_consume
+            this._connection.config.transport.warn(loggerAlias, null);
+            this._connection.config.logger.warn({
+              message: `${loggerAlias} Consumer was cancelled by server for queue '${queue}'`,
+              error: null,
+              params: { queue },
+            });
+            return;
+          }
+
+          const messageString = msg.content.toString();
+          this._connection.config.transport.debug(loggerAlias, `[${queue}] < ${messageString}`);
+          this._connection.config.logger.debug({
+            message: `${loggerAlias} [${queue}] < ${messageString}`,
+            params: { queue, message: messageString },
+          });
+
+          // main answer management chaining
+          // receive message, parse it, execute callback, check if should answer, ack/reject message
+          Promise.resolve(parsers.in(msg))
+            .then((body) => callback(body, msg.properties))
+            .then(this.checkRpc(msg, queue))
+            .then(() => {
+              try {
+                // This is synchronous
```

Comment:
I think this comment is misleading, this method is not sync, it's async. It just don't return promise/callback.

Reply:
Removed the comment.